src/core/utils.cpp

```diff
@@ -427,4 +429,29 @@ generateModule(StringRef Routine, const Triple &Triple, StringRef Extension,
   }
 }
 
+IRChangesMonitor::IRChangesMonitor(const llvm::Module &M,
+                                   llvm::StringRef PassName)
+    : Mod(M) {
+  ObfuscationConfig *UserConfig = PyConfig::instance().getUserConfig();
+  if (UserConfig->has_report_diff_override()) {
+    this->UserConfig = UserConfig;
+    this->PassName = PassName.str();
+    llvm::raw_string_ostream(OriginalIR) << Mod;
+  }
+}
+
+PreservedAnalyses IRChangesMonitor::report() {
+  if (UserConfig) {
+    std::string ObfuscatedIR;
+    llvm::raw_string_ostream(ObfuscatedIR) << Mod;
+    if (OriginalIR != ObfuscatedIR) {
+      assert(!ChangeReported &&
+             "Textual IR change detected that transformation didn't report");
```

Comment:
```suggestion
      assert(ChangeReported &&
             "Textual IR change detected that transformation didn't report");
```
Other way around, right?